src/mower_logic/src/mower_logic/mower_logic.cpp

```diff
@@ -374,20 +374,24 @@ void checkSafety(const ros::TimerEvent &timer_event) {
     const auto status_time = getStatusTime();
     const auto last_good_gps = getLastGoodGPS();
 
-
     // call the mower
     setMowerEnabled(currentBehavior != nullptr && currentBehavior->mower_enabled());
 
     high_level_status.emergency = last_status.emergency;
     high_level_status.is_charging = last_status.v_charge > 10.0;
 
     // send to idle if emergency and we're not recording
-    if(last_status.emergency) {
-        if(currentBehavior != &AreaRecordingBehavior::INSTANCE && currentBehavior != &IdleBehavior::INSTANCE) {
-            abortExecution();
-        } else if(last_status.v_charge > 10.0) {
-            // emergency and docked and idle or area recording, so it's safe to reset the emergency mode, reset it. It's safe since we won't start moving in this mode.
-            setEmergencyMode(false);
+    if (currentBehavior != nullptr) {
+        if(last_status.emergency) {
+            currentBehavior->requestPause();
+            if(currentBehavior == &AreaRecordingBehavior::INSTANCE || currentBehavior == &IdleBehavior::INSTANCE) {
+                if(last_status.v_charge > 10.0) {
+                    // emergency and docked and idle or area recording, so it's safe to reset the emergency mode, reset it. It's safe since we won't start moving in this mode.
+                    setEmergencyMode(false);
+                }
+            }
+        } else {
+            currentBehavior->requestContinue();
```

Comment:
regular pause/continue (non-emergency) no longer works though, this will always continue immediately if there is no emergency

Reply:
Fixed in latest commit